cart/application/cartService.go

```diff
@@ -861,3 +861,33 @@ func (cs *CartService) dispatchAllEvents(ctx context.Context, events []flamingo.
 		cs.eventRouter.Dispatch(ctx, e)
 	}
 }
+
+// CheckAndAdjustStockInCart - checks if the stock of an item in the cart is lower than the quantity of it and if so updates the quantity
+func (cs *CartService) CheckAndAdjustStockInCart(ctx context.Context, session *web.Session) error {
```

Comment:
Stock is a "concept" that was not known to the cart.
The cart has the concept of "QtyRestrictors"  and we should use this as naming.
e.g. AdjustToRestrictedQty

The result of that function should be a value object that descibes what has been adjusted (e.g. slice of AdjustmentInfo (cartid, old, new qty, restrictor?)